py/stream.c

```diff
@@ -303,7 +303,12 @@ static mp_obj_t stream_readinto(size_t n_args, const mp_obj_t *args) {
     }
 
     int error;
-    mp_uint_t out_sz = mp_stream_read_exactly(args[0], bufinfo.buf, len, &error);
+    mp_uint_t out_sz;
+    if (flags & MP_STREAM_RW_ONCE) {
```

Comment:
This `if` can be simplified to a single line:
```
mp_stream_rw(args[0], bufinfo.buf, len, &error, flags);
```
and then pass in `MP_STREAM_RW_READ` as flags from `stream_readinto()`, and `MP_STREAM_RW_READ | MP_STREAM_RW_ONCE` from `stream_readinto1()`.